public/app/features/integrated-alerting/components/NotificationChannel/AddNotificationChannelModal/PagerDutyFields/PagerDutyFields.tsx

```diff
@@ -1,18 +1,38 @@
 import React, { FC } from 'react';
-import { TextInputField, validators } from '@percona/platform-core';
+import { RadioButtonGroupField, TextInputField, validators } from '@percona/platform-core';
 import { Messages } from '../AddNotificationChannelModal.messages';
 import { PagerDutyFieldsProps } from './PagerDutyFields.types';
+import { PagerDutyKeyType } from '../../NotificationChannel.types';
+import { PAGER_DUTY_TYPE_OPTIONS } from '../AddNotificationChannel.constants';
 
-export const PagerDutyFields: FC<PagerDutyFieldsProps> = ({ values }) => {
-  const { required } = validators;
-  const { routing, service } = values;
-  const routingValidators = service ? [] : [required];
-  const serviceValidators = routing ? [] : [required];
+const { required } = validators;
+const keyValidator = [required];
 
+export const PagerDutyFields: FC<PagerDutyFieldsProps> = ({ values, mutators }) => {
   return (
     <>
-      <TextInputField name="routing" label={Messages.fields.routingKey} validators={routingValidators} />
-      <TextInputField name="service" label={Messages.fields.serviceKey} validators={serviceValidators} />
+      <RadioButtonGroupField
+        inputProps={{
+          /**
+           * TODO We probably should change platform-core to allow to pass values to the options.
+           * That way, we can take the value from the event itself here and call the mutator, rather than checking the form values.
+           */
+          onInput: () => {
+            values?.keyType === PagerDutyKeyType.service
+              ? mutators?.resetKey(PagerDutyKeyType.routing)
+              : mutators?.resetKey(PagerDutyKeyType.service);
+          },
+        }}
+        name="keyType"
+        options={PAGER_DUTY_TYPE_OPTIONS}
+        initialValue={values?.keyType || PAGER_DUTY_TYPE_OPTIONS[0].value}
```

Comment:
Not sure why we need `values?.keyType`. Is not enough to have the default initial value always?

Reply:
Right, in the context of the form, we don't need this initialisation here because the form't state will just fill it correctly. But when used isolated, like when testing,  it should be initialised with either the value thats coming from the props or just the first option we can show.